functests/4_latency/latency.go

```diff
@@ -249,22 +251,36 @@ var _ = Describe("[performance] Latency Test", func() {
 
 func getLatencyTestRun() (bool, error) {
 	if latencyTestRunEnv, ok := os.LookupEnv("LATENCY_TEST_RUN"); ok {
-		return strconv.ParseBool(latencyTestRunEnv)
+		if val, err := strconv.ParseBool(latencyTestRunEnv); err != nil {
+			return val, fmt.Errorf("the environment variable LATENCY_TEST_RUN has incorrect value %q: %w", latencyTestRunEnv, err)
+		} else {
+			return val, nil
+		}
 	}
 	return defaultTestRun, nil
 }
 
-func getLatencyTestRuntime() string {
+func getLatencyTestRuntime() (string, error) {
 	if latencyTestRuntimeEnv, ok := os.LookupEnv("LATENCY_TEST_RUNTIME"); ok {
-		return latencyTestRuntimeEnv
+		if val, err := strconv.Atoi(latencyTestRuntimeEnv); err != nil || val < 1 {
```

Comment:
The logic here is a bit confusing, I would separate that, so:
```
val, err := strconv.Atoi(latencyTestRuntimeEnv)
if err != nil {
    return latencyTestRuntimeEnv, fmt.Errorf("the environment variable LATENCY_TEST_RUNTIME has incorrect value %q, it must be a positive integer with maximum value of %d", latencyTestRuntimeEnv, MAX_INT)
  }
  
if val < 1 {
    return "", fmt.Errorf("the environment variable LATENCY_TEST_RUNTIME has a nonpositive value %q, it must be a positive integer with maximum value of %d", latencyTestRuntimeEnv, MAX_INT)
}

return latencyTestRuntimeEnv, nil
```

Reply:
done , thanks!